openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -116,137 +116,59 @@ public OpenAPI modifyWithBallerinaConventions(OpenAPI openapi, Map<String, Strin
-                                                       Map<String, ApiResponse> modifiedResponses) {
-        if (components.getResponses() != null) {
-            Map<String, ApiResponse> responsesEntry = components.getResponses();
-            for (Map.Entry<String, ApiResponse> response : responsesEntry.entrySet()) {
-                ApiResponse modifiedResponse = updateSchemaTypeForResponses(schemaName, modifiedName,
-                        response.getValue());
-                modifiedResponses.put(response.getKey(), modifiedResponse);
+        String openApiJson = Json.pretty(openapi);
+        for (Map.Entry<String, String> entry : nameMap.entrySet()) {
+            String schemaName = entry.getKey();
+            String modifiedName = entry.getValue();
+            if (schemaName.equals(modifiedName)) {
+                continue;
             }
-        }
-    }
+            String patternString = ("(\\\"\\$ref\\\"\\s*:\\s*\\\"#/components/schemas/%s\\\")|" +
+                    "('\\$ref'\\s*:\\s*'#/components/schemas/%s')").formatted(schemaName, schemaName);
+            Pattern pattern = Pattern.compile(patternString);
+            Matcher matcher = pattern.matcher(openApiJson);
 
-    private static void updateComponentRequestBodySection(Components components, String schemaName,
-                                                          String modifiedName,
-                                                          Map<String, RequestBody> modifiedRequestBodies) {
-        if (components.getRequestBodies() != null) {
-            for (Map.Entry<String, RequestBody> requestBody : components.getRequestBodies().entrySet()) {
-                RequestBody requestBodyValue = updateRequestBodySchemaType(schemaName, modifiedName,
-                        requestBody.getValue());
-                modifiedRequestBodies.put(requestBody.getKey(), requestBodyValue);
-            }
+            String replacement = "\"$ref\":\"#/components/schemas/%s\"".formatted(modifiedName).replace("$", "\\$");
+            openApiJson = matcher.replaceAll(replacement);
         }
-    }
 
-    private static void updateComponentSchemaSection(Map<String, Schema> schemas, String schemaName,
-                                                     String modifiedName, Map<String, Schema> modifiedSchemas) {
-        for (Map.Entry<String, Schema> componentSchema : schemas.entrySet()) {
-            Schema value = componentSchema.getValue();
-            updateSchemaWithReference(schemaName, modifiedName, value);
-            modifiedSchemas.put(modifiedName, value);
+        ParseOptions parseOptions = new ParseOptions();
+        SwaggerParseResult parseResult = new OpenAPIParser().readContents(openApiJson, null, parseOptions);
+        if (!parseResult.getMessages().isEmpty()) {
+            outStream.println("Sanitized OpenAPI contains errors:");
+            parseResult.getMessages().forEach(outStream::println);
```

Comment:
The logic is refactored to replace the string content of the OpenAPI model. For that reason, we have converted the OpenAPI model to string(in json format), do the replacing and convert it back to the OpenAPI model. This additionally ensures that the modified OpenAPI specification has no errors.

But here when we get any errors while parsing the modified OpenAPI json string to OpenAPI model, we print those errors and **anyway continue the code generation using the modified OpenAPI model**. Is that fine? Or should we fall-back to the original OpenAPI specification? If so we need to make a copy of the OpenAPI model before modifying. @daneshk @lnash94 FYI

Reply:
What scenarios in which we end up with an error when converting back? Is it okay to continue code generation when there is an error in modified model?

Reply:
If the user can recover from the generation failure or compilation error by modifying the sanitized schema. I am +1 for continuing, as we give enough context to recover.

Reply:
In the code generation process, we are going through the same parser issue check before reaching the modifier. Therefore, I prefer to continue code generation with the user-provided OAS when there is an issue after the modifier. Additionally, I'm in favor of adding a warning message to inform the user that we did not apply the Ballerina set of rules for the given OAS.